Basic dev docs for VMSS processor (#498)
microsoft/AzureTRE: processor_function/vm_porter/readme.md

docker run -it -v /var/run/docker.sock:/var/run/docker.sock  --env-file .env  rp

## Local development

To work locally checkout the source code and run

``pip install -r requirements.txt``

If you use visual studio code you can set up your launch.json to include the follwing block which will enable launching and debugging.

```json
{
      "name": "VMSS Processor",
      "type": "python",
      "request": "launch",
      "program": "vm_porter/runner.py",
      "console": "integratedTerminal",
      "cwd": "${workspaceFolder}/processor_function",
      "env": {
        "PYTHONPATH": ".",
        "AZURE_CLIENT_ID": "",
        "AZURE_CLIENT_SECRET": "",
        "AZURE_TENANT_ID": "",
        "REGISTRY_SERVER": "",
        "ARM_SUBSCRIPTION_ID": "",
        "TERRAFORM_STATE_CONTAINER_NAME": "",
        "MGMT_RESOURCE_GROUP_NAME": "",
        "MGMT_STORAGE_ACCOUNT_NAME": "",
        "SERVICE_BUS_DEPLOYMENT_STATUS_UPDATE_QUEUE": "deploy",
        "SERVICE_BUS_RESOURCE_REQUEST_QUEUE": "",
        "SERVICE_BUS_FULLY_QUALIFIED_NAMESPACE": "",
        "ARM_CLIENT_ID": "",
        "ARM_CLIENT_SECRET": "",
        "ARM_TENANT_ID": ""
      }
```

> As you can see, to work locally we use a service princiapl(SP). This SP needs enough permissions to be able to talke to service bus and deploy resources when used within porter. That means the service principal needs Owner access to subscription(ARM_SUBSCRIPTION_ID) and also needs **Azure Service Bus Data Sender** and **Azure Service Bus Data Receiver** on the service bus namespace defined above(SERVICE_BUS_FULLY_QUALIFIED_NAMESPACE).

Reviewer: ```suggestion
When working locally we use a service principal (SP). This SP needs enough permissions to be able to talk to service bus and to deploy resources into the subscription. That means the service principal needs Owner access to subscription(ARM_SUBSCRIPTION_ID) and also needs **Azure Service Bus Data Sender** and **Azure Service Bus Data Receiver** on the service bus namespace defined above(SERVICE_BUS_FULLY_QUALIFIED_NAMESPACE).
```
